object/object_test.go

```diff
@@ -1327,6 +1327,388 @@ func TestUploadService_UploadAbort(t *testing.T) {
+	}
+	type args struct {
+		ctx    aws.Context
+		bucket *string
+		keys   []*string
+	}
+	tests := []struct {
+		name        string
+		fields      fields
+		args        args
+		wantSuccess []string
+		wantFailed  []string
+		wantErr     bool
+	}{
+		{
+			name: "delete only one object",
+			fields: fields{
+				s3Client: s3Client,
+			},
+			args: args{
+				keys:   []*string{key1},
+				bucket: aws.String("testbucket"),
+				ctx:    context.Background(),
+			},
+			wantSuccess: []string{*key1},
+			wantFailed:  []string{},
+			wantErr:     false,
+		},
+		{
+			name: "delete two objects",
+			fields: fields{
+				s3Client: s3Client,
+			},
+			args: args{
+				keys:   []*string{key2, key3},
+				bucket: aws.String("testbucket"),
+				ctx:    context.Background(),
+			},
+			wantSuccess: []string{*key2, *key3},
+			wantFailed:  []string{},
+			wantErr:     false,
+		},
+		{
+			name: "delete valid and invalid objects",
+			fields: fields{
+				s3Client: s3Client,
+			},
+			args: args{
+				keys:   []*string{key4, aws.String("oneoneone")},
+				bucket: aws.String("testbucket"),
+				ctx:    context.Background(),
+			},
+			wantSuccess: []string{*key4, "oneoneone"},
+			wantFailed:  []string{},
+			wantErr:     false,
+		},
+		{
+			name: "delete invalid object",
+			fields: fields{
+				s3Client: s3Client,
+			},
+			args: args{
+				keys:   []*string{aws.String("oneoneone")},
+				bucket: aws.String("testbucket"),
+				ctx:    context.Background(),
+			},
+			wantSuccess: []string{"oneoneone"},
+			wantFailed:  []string{},
+			wantErr:     false,
+		},
+		{
+			name: "delete nil keys",
+			fields: fields{
+				s3Client: s3Client,
+			},
+			args: args{
+				keys:   nil,
+				bucket: aws.String("testbucket"),
+				ctx:    context.Background(),
+			},
+			wantSuccess: []string{},
+			wantFailed:  []string{},
+			wantErr:     true,
+		},
+		{
+			name: "delete nil bucket",
+			fields: fields{
+				s3Client: s3Client,
+			},
+			args: args{
+				keys:   []*string{aws.String("valid")},
+				bucket: nil,
+				ctx:    context.Background(),
+			},
+			wantSuccess: []string{},
+			wantFailed:  []string{},
+			wantErr:     true,
+		},
+		{
+			name: "delete empty bucket",
+			fields: fields{
+				s3Client: s3Client,
+			},
+			args: args{
+				keys:   []*string{aws.String("valid")},
+				bucket: aws.String(""),
+				ctx:    context.Background(),
+			},
+			wantSuccess: []string{},
+			wantFailed:  []string{},
+			wantErr:     true,
+		},
+		{
+			name: "delete empty key",
+			fields: fields{
+				s3Client: s3Client,
+			},
+			args: args{
+				keys:   []*string{},
+				bucket: aws.String("testbucket"),
+				ctx:    context.Background(),
+			},
+			wantSuccess: []string{},
+			wantFailed:  []string{},
+			wantErr:     true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := object.NewService(tt.fields.s3Client)
+			got, err := s.DeleteObjects(tt.args.ctx, tt.args.bucket, tt.args.keys)
+			if (err != nil) != tt.wantErr {
```

Comment:
maybe change to `(err == nil) == tt.wantErr`?
or get the `err != nil` inside a variable so it will be more readable.